Extensions/Oxide.CSharp/PluginCompiler.cs

```diff
@@ -494,5 +494,34 @@ private static string GetHash(string filePath, HashAlgorithm algorithm)
                 return BitConverter.ToString(hash).Replace("-", string.Empty).ToLower();
             }
         }
+
+        private static bool FirewallBlocking()
+        {
+            try
+            {
+                // we can use any url below that contains a CSharp source file.
+                var request =
+                    (HttpWebRequest)WebRequest.Create(
+                        $"https://devdrop.blob.core.windows.net/downloads/FirewallTest.cs");
+                var response = (HttpWebResponse)request.GetResponse();
+                var statusCode = (int)response.StatusCode;
+                switch (statusCode)
+                {
+                    case 401:
```

Comment:
I don't see how checking for a valid response code would test if web requests are being blocked. If anything, you'd want no response or a response of 0.

Reply:
I tested using squid & squidguard on my PFSense firewall. It gave 403 when denied, or 401 when set to required login/password for bypass. Other firewalls/content filters may display different behaviors.